### PR TITLE
Site Editor Tracking: Add events for selecting content items in the navigation sidebar

### PR DESCRIPTION
**Pull request by david-szabo97**
#### Changes proposed in this Pull Request

* Track selecting content item in the navigation sidebar
  * Event name: `wpcom_block_editor_nav_sidebar_item_edit`
  * Event properties:
    * `item_type` - type of the selected item (page, post, category, etc.)
    * `item_slug` - slug of the selected item 

Note `setPage`is called twice due to a bug in Gutenberg: WordPress/gutenberg#33286

#### Testing instructions

* Open Site Editor
* Open navigation sidebar
* Change to a content item
* Make sure event is tracked

Make sure E2E tests are passing.

Related to #

### CODE REVIEW

File: apps/wpcom-block-editor/src/wpcom/features/tracking.js [hunk 1]
```diff
@@ -460,6 +460,13 @@ const trackSiteEditorChangeTemplatePart = ( id ) => {
 	} );
 };
 
+const trackSiteEditorChangeContent = ( { type, slug } ) => {
+	tracksRecordEvent( 'wpcom_block_editor_nav_sidebar_item_edit', {
+		item_type: type,
+		item_slug: slug,
```
Comment (Addison-Stavlo) [inline apps/wpcom-block-editor/src/wpcom/features/tracking.js:508]: It seems when we select a 'category' page we just get `item_slug` with the category name and no `item_type`.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js [hunk 2]
```diff
@@ -975,6 +975,28 @@ describe( `[${ host }] Calypso Gutenberg Site Editor Tracking: (${ screenSize })
...
+					'[role="region"][aria-label="Navigation Sidebar"] [role="menu"] [title="Home"] button'
+				);
+				await driverHelper.clickWhenClickable( this.driver, contentMenuItemLocator );
+
+				const eventsStack = await getEventsStack( this.driver );
+				const editEvents = eventsStack.filter(
+					( [ eventName ] ) => eventName === 'wpcom_block_editor_nav_sidebar_item_edit'
+				);
+				// TODO: Change this to 1 once https://github.com/WordPress/gutenberg/pull/33286 fix is deployed
+				assert.strictEqual( editEvents.length, 2 );
```
Comment (Addison-Stavlo) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js]: Another idea might be to debounce the function that sends the event with a very small time window so only 1 event sends.  Although, this double-dispatch should definitely be fixed anyways and will solve this problem.

**Review by Addison-Stavlo — COMMENTED** [review]
This is looking good so far and tests are working well.  just a couple comments.

File: apps/wpcom-block-editor/src/wpcom/features/tracking.js (hunk 1 above)
Comment (david-szabo97) [reply apps/wpcom-block-editor/src/wpcom/features/tracking.js:508]: We don't have that information from the args. 😢  Used a selector to check the active menu. Type is determined based on that in this case.

**Comment by david-szabo97** [comment]
Anyone planning to run e2e tests, make sure to comment this line:

https://github.com/Automattic/wp-calypso/blob/580ee065bcc92f12ab58a9a8202c58f98b7a8b6b/test/e2e/lib/components/site-editor-component.js#L250

Latest Gutenberg version introduced a regression, showing the snackbar below the inserter, which prevents us from clicking on the notice to dismiss it.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js (hunk 2 above)
Comment (mattwiebe) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js]: [Debouncing!](https://github.com/Automattic/wp-calypso/pull/54416)

**Comment by mattwiebe** [comment]
Is it expected that `wpcom_block_editor_nav_sidebar_item_edit` would be called upon the Site Editor upon first load?

```
{
  "editor_type": "site",
  "item_type": "template",
  "item_id": "pub/tt1-blocks//index",
  "blog_id": 34321322,
  "site_type": "simple",
  "user_locale": "en"
}
```
---

The rest of this may be totally expected but I haven't been following the stats work so I had some questions about expected behaviour. Since I was debugging on `wpcom-block-editor:analytics:tracks`, I noticed that after I clicked on a page to edit within the Site Editor ( `wpcom_block_editor_nav_sidebar_item_edit` 👍 ) that I got some other events that may or may not be expected?

Screen Shot 2021-07-12 at 15 38 03

Basically what happens is that, after the page is selected, Gutenberg repaints its editor canvas with the blocks of the selected page. This winds up callind `wpcom_block_inserted` even though these aren't being inserted explicitly by the user, so that could muddy our stats a bit.

**Comment by david-szabo97** [comment]
> Is it expected that wpcom_block_editor_nav_sidebar_item_edit would be called upon the Site Editor upon first load?

Nope! Fixed it. Though it's quite hacky 😬 

> The rest of this may be totally expected but I haven't been following the stats work so I had some questions about expected behaviour. Since I was debugging on wpcom-block-editor:analytics:tracks, I noticed that after I clicked on a page to edit within the Site Editor ( wpcom_block_editor_nav_sidebar_item_edit 👍 ) that I got some other events that may or may not be expected?

This is a known issue 😞  We have a workaround for reusable blocks and template parts. But we found out this problem comes up with many other blocks that includes/loads inner blocks.

**Review by mattwiebe — APPROVED** [review]
Thanks for the fixes, this looks good and tests properly 🚢
